src/cryptography/x509/extensions.py

```diff
@@ -134,38 +137,60 @@ def get_extension_for_class(self, extclass) -> "Extension[ExtensionType]":
             if isinstance(ext.value, extclass):
                 return ext
 
+        oid = typing.cast(ObjectIdentifier, extclass.oid)
```

Comment:
Same issue here as in #5830: ExtensionType.oid is an abstract property (= an abstract _instance_ property), but of course all classes define a class attribute. Should we replace with 

```
class ExtensionType:
     oid: ObjectIdentifier
```

?

Reply:
Will doing `oid: ObjectIdentifier` work with the fact that it's a classvar on most `ExtensionType` implementors, but an instancevar on `UnknownExtension`?

Reply:
Just tested this, and it seems so, at least from a mypy point of view, here's the diff I tested:

```diff
 class ExtensionType(metaclass=abc.ABCMeta):
-    @abc.abstractproperty
-    def oid(self) -> ObjectIdentifier:
-        """
-        Returns the oid associated with the given extension type.
-        """
+    oid: ObjectIdentifier
```

HOWEVER I must point out - from [abc](https://docs.python.org/3/library/abc.html):

>  A class that has a metaclass derived from ABCMeta cannot be instantiated unless all of its abstract methods and properties are overridden.

That means, the change would mean you can instantiate an ExtensionType, if you want to:

```
>>> x509.ExtensionType()
<cryptography.x509.extensions.ExtensionType object at 0x...>
```

Reply:
Unfortunate, but not a deal breaker. I guess it's do it!